lib/src/engines/jael.dart

```diff
@@ -1,24 +1,49 @@
-import 'dart:io';
 import 'package:dart_express/src/engines/engine.dart';
-import 'package:code_buffer/code_buffer.dart' deferred as jael_code_buffer;
-import 'package:jael/jael.dart' deferred as jael;
-import 'package:symbol_table/symbol_table.dart' deferred as jael_symbol_table;
+import 'package:code_buffer/code_buffer.dart';
+import 'package:file/local.dart';
+import 'package:jael/jael.dart' as jael;
+import 'package:jael_preprocessor/jael_preprocessor.dart' as jael;
+import 'package:symbol_table/symbol_table.dart';
 
 class JaelEngine {
-  static String ext = '.jl';
+  static String ext = '.jael';
 
   static handler(
       String filePath, Map<String, dynamic> options, HandlerCallback callback) {
-    var file = File.fromUri(Uri.file(filePath));
+    var fs = LocalFileSystem();
+    var file = fs.file(filePath);
 
-    file.readAsString().then((str) {
-      var buffer = jael_code_buffer.CodeBuffer();
-      var document = jael.parseDocument(str, sourceUrl: filePath);
-      var scope = jael_symbol_table.SymbolTable(values: options);
+    file.readAsString().then((str) async {
+      // Store any parsing/resolution/rendering errors
+      var errors = <jael.JaelError>[];
 
-      jael.Renderer().render(document, buffer, scope);
+      // Parse the document, of course.
+      var buffer = CodeBuffer();
+      var document =
+          jael.parseDocument(str, sourceUrl: filePath, onError: errors.add);
 
-      callback(null, buffer.toString());
+      // Resolve template includes, etc.
+      document = await jael.resolve(document, fs.directory('views'),
```

Comment:
I'm not 100% sure why the `jael.resolve` method needs the directory, so this might be a stupid question, but the `filePath` should be the full file path to the view that the user is trying to render and it _might_ not be in the `views` folder if they specified a different `viewsPath` [here](https://github.com/deriegle/dart-express/blob/master/lib/src/app.dart#L14).

Will this be an issue if the file they are trying to render is not located in the `views` folder?

cc @thosakwe